Mathlib/RepresentationTheory/Homological/GroupHomology/Basic.lean

```diff
@@ -0,0 +1,217 @@
+is only defined for commutative rings.
+
+## TODO
+
+* API for homology in low degree: $\mathrm{H}_0, \mathrm{H}_1$ and $\mathrm{H}_2.$ For example,
+the corestriction-coinflation exact sequence.
+* The long exact sequence in homology attached to a short exact sequence of representations.
+* Upgrading `groupHomologyIsoTor` to an isomorphism of derived functors.
+
+-/
+
+noncomputable section
+
+universe u
+
+open CategoryTheory CategoryTheory.Limits
+
+namespace Rep
+
+variable (k G : Type u) [CommRing k] [Group G]
+
+open MonoidalCategory Representation Finsupp
+
+/-- The left-derived functors given by deriving the second argument of `A, B ↦ (A ⊗[k] B)_G`. -/
+@[simps]
+def Tor (n : ℕ) : Rep k G ⥤ Rep k G ⥤ ModuleCat k where
+  obj X := Functor.leftDerived ((coinvariantsTensor k G).obj X) n
+  map f := NatTrans.leftDerived ((coinvariantsTensor k G).map f) n
+
+variable {k G} (A : Rep k G)
+
+/-- The higher `Tor` groups for `X` and `Y` are zero if `Y` is projective. -/
+lemma isZero_Tor_succ_of_projective (X Y : Rep k G) [Projective Y] (n : ℕ) :
+    IsZero (((Tor k G (n + 1)).obj X).obj Y) :=
+  Functor.isZero_leftDerived_obj_projective_succ ..
+
+/-- Given a `k`-linear `G`-representation `A`, this is the chain complex `(A ⊗[k] P)_G`, where
+`P` is the bar resolution of `k` as a trivial representation. -/
+abbrev coinvariantsTensorBarResolution [DecidableEq G] :=
+  (((coinvariantsTensor k G).obj A).mapHomologicalComplex _).obj (barComplex k G)
+
+end Rep
+
+namespace groupHomology
+
+open Rep Finsupp
+
+variable {k G : Type u} [CommRing k] [Group G] (A : Rep k G) (n : ℕ)
+
+namespace inhomogeneousChains
+
+/-- The differential in the complex of inhomogeneous chains used to calculate group homology. -/
+def d : ModuleCat.of k ((Fin (n + 1) → G) →₀ A) ⟶ ModuleCat.of k ((Fin n → G) →₀ A) :=
+  ModuleCat.ofHom <| lsum (R := k) k fun g => lsingle (fun i => g i.succ) ∘ₗ A.ρ (g 0)⁻¹ +
+    Finset.univ.sum fun j : Fin (n + 1) =>
+      (-1 : k) ^ ((j : ℕ) + 1) • lsingle (Fin.contractNth j (· * ·) g)
+
+variable {A n} in
+@[simp]
+theorem d_single (n : ℕ) (g : Fin (n + 1) → G) (a : A) :
+    d A n (single g a) = single (fun i => g i.succ) (A.ρ (g 0)⁻¹ a) +
+      Finset.univ.sum fun j : Fin (n + 1) =>
+        (-1 : k) ^ ((j : ℕ) + 1) • single (Fin.contractNth j (· * ·) g) a := by
+  simp [d]
+
+open ModuleCat.MonoidalCategory
+
+theorem d_eq [DecidableEq G] :
+    d A n = (coinvariantsTensorFreeLEquiv A (Fin (n + 1) → G)).toModuleIso.inv ≫
+      (coinvariantsTensorBarResolution A).d (n + 1) n ≫
+      (coinvariantsTensorFreeLEquiv A (Fin n → G)).toModuleIso.hom := by
+  ext : 3
+  simp [d_single (k := k) (G := G), finsuppToCoinvariantsTensorFree_single (A := A),
+    coinvariantsTensorFreeToFinsupp_mk_tmul_single (A := A) (α := Fin n → G),
+    ModuleCat.MonoidalCategory.tensorObj, ModuleCat.MonoidalCategory.whiskerLeft, tensorObj_def,
+    whiskerLeft_def, TensorProduct.tmul_add, TensorProduct.tmul_sum, barComplex.d_single (k := k)]
```

Comment:
```suggestion
  simp [d_single (k := k), ModuleCat.MonoidalCategory.tensorObj,
    ModuleCat.MonoidalCategory.whiskerLeft, tensorObj_def, whiskerLeft_def, TensorProduct.tmul_add,
    TensorProduct.tmul_sum, barComplex.d_single (k := k)]
```
(it's just a proper subset of what you wrote)

Reply:
Done, thanks